app/src/main/java/com/example/ithaca_transit_android_v2/models/Direction.kt

```diff
@@ -1,12 +1,21 @@
 package com.example.ithaca_transit_android_v2.models
-import java.util.*
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+import java.util.Date
 
+@JsonClass(generateAdapter = true)
 data class Direction(
+    val type : DirectionType,
+    @Json(name = "path")
     val listOfCoordinates: List<Coordinate>,
     val startTime: Date,
     val endTime: Date,
-    val startLocation: Location,
-    val endLocation: Location,
-    val busStops: List<Location>,
-    val busNumber: Int
+    @Json(name = "startLocation")
+    val startCoords: Coordinate,
+    @Json(name = "endLocation")
+    val endCoords: Coordinate,
+    @Json(name = "stops")
+    val busStops: List<Stop>,
+    @Json(name = "routeNumber")
+    val busNumber: Int?
```

Comment:
Is there a way that we could prevent getting a null using an adapter?

Reply:
I think we can leave it as nullable for now. When displaying the bus number, we'll need to check if busNumber is null and don't display it in that case (the only occasion when bus number is null should be a walking direction).